pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -389,3 +392,30 @@ func (p *Provider) getInstanceProfile(nodeClass *v1beta1.EC2NodeClass) (string,
 	}
 	return "", errors.New("neither spec.instanceProfile or spec.role is specified")
 }
+
+func (p *Provider) DeleteLaunchTemplates(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) error {
+	clusterName := options.FromContext(ctx).ClusterName
+	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).With("tag-key", karpenterManagedTagKey, "tag-value", clusterName))
+	var deleteLTerr error
+	describeErr := p.ec2api.DescribeLaunchTemplatesPagesWithContext(ctx, &ec2.DescribeLaunchTemplatesInput{
+		Filters: []*ec2.Filter{{Name: aws.String(fmt.Sprintf("tag:%s", karpenterManagedTagKey)), Values: []*string{aws.String(clusterName)}},
+			{Name: aws.String(fmt.Sprintf("tag:%s", v1beta1.LabelNodeClass)), Values: []*string{aws.String(nodeClass.Name)}}},
+	}, func(output *ec2.DescribeLaunchTemplatesOutput, _ bool) bool {
```

Comment:
Super optional nit, but this is pretty dense and a little hard for me to read so a small formatting suggestion:
```suggestion
	describeErr := p.ec2api.DescribeLaunchTemplatesPagesWithContext(ctx, &ec2.DescribeLaunchTemplatesInput{
		Filters: []*ec2.Filter{
			{Name: aws.String(fmt.Sprintf("tag:%s", karpenterManagedTagKey)), Values: []*string{aws.String(clusterName)}},
			{Name: aws.String(fmt.Sprintf("tag:%s", v1beta1.LabelNodeClass)), Values: []*string{aws.String(nodeClass.Name)}}
		},
	}, func(output *ec2.DescribeLaunchTemplatesOutput, _ bool) bool {
```